indico/htdocs/js/utils/forms.js

```diff
@@ -64,8 +64,13 @@
         forms.find('[data-disabled-until-change]').prop('disabled', true);
         forms.each(function() {
             var $this = $(this);
-            $this.data('initialData', $this.serialize());
-            $this.data('fieldsChanged', false);
+            function _resetData() {
+                $this.data('initialData', $this.serialize());
+                $this.data('fieldsChanged', false);
+            }
+            _resetData();
```

Comment:
i'd trigger the change event on the form here (or even move it into a function and just call it) instead of when calling the fieldsSaved signal